scanners/ios/icon.go

```diff
@@ -12,54 +12,29 @@ import (
 	"github.com/bitrise-io/go-xcode/xcodeproject/xcscheme"
 )
 
-// lookupIconBySchemeName returns possible ios app icons for a scheme.
-func lookupIconBySchemeName(projectPath string, schemeName string, basepath string) (models.Icons, error) {
-	project, err := xcodeproj.Open(projectPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to open project file: %s, error: %s", projectPath, err)
-	}
-
-	scheme, _, err := project.Scheme(schemeName)
-	if err != nil {
-		return nil, fmt.Errorf("failed to find scheme (%s) in project (%s), error: %s", schemeName, project.Path, err)
-	}
-
-	blueprintID := getBlueprintID(*scheme)
-	if blueprintID == "" {
-		log.TDebugf("scheme (%s) does not contain app buildable reference in project (%s)", scheme.Name, project.Path)
-		return nil, nil
+func getMainTarget(project xcodeproj.XcodeProj, scheme xcscheme.Scheme) (xcodeproj.Target, error) {
+	entry, found := scheme.AppBuildActionEntry()
+	if !found {
+		return xcodeproj.Target{}, fmt.Errorf("scheme (%s) does not contain app buildable reference in project (%s)", scheme.Name, project.Path)
 	}
 
-	// Search for the main target
-	mainTarget, found := targetByBlueprintID(project.Proj.Targets, blueprintID)
+	blueprintID := entry.BuildableReference.BlueprintIdentifier
+	mainTarget, found := project.Proj.Target(blueprintID)
 	if !found {
-		return nil, fmt.Errorf("no target found for blueprint ID (%s) project (%s)", blueprintID, project.Path)
+		return xcodeproj.Target{}, fmt.Errorf("no target found for blueprint ID (%s) project (%s)", blueprintID, project.Path)
```

Comment:
`no target found for blueprint ID (%s) project (%s)` -> `no target found for blueprint ID (%s) in project (%s)`

Reply:
fixed